src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,127 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int capacity;
+    private final float loadFactor;
+    private int threshold;
+
+    private V nullKeyValue = null;
+    private boolean nullKeyPresent = false;
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
```

Comment:
The inner class 'Node' should not have public getters and setters as we have direct access to its private fields within the outer class 'MyHashMap'.